src/cudnn/safe/core.rs

```diff
@@ -32,8 +31,11 @@ impl Cudnn {
     /// write to the same memory address.
     pub unsafe fn set_stream(&self, opt_stream: Option<&CudaStream>) -> Result<(), CudnnError> {
         match opt_stream {
-            Some(s) => result::set_stream(self.handle, s.stream as *mut _),
-            None => result::set_stream(self.handle, self.device.stream as *mut _),
+            Some(s) => {
+                assert_eq!(self.device.ordinal(), s.device.ordinal());
```

Comment:
TODO should this assert or return an error? is setting the stream to a different device valid? Same comment for other set_stream methods